### ISSUE TITLE
wrong address should fail gracefully

### ISSUE THREAD

**Issue by fallerOfFalls** [clemahieu/raiblocks#274]
Every cryptocurrency I know of has this problem. If you somehow input the address incorrectly, it doesn't warn you, it just sends it to an address nobody owns. This means every send is stressful, and this stress makes it unappealing to use.

In other respects (feeless, instant) RaiBlocks is like cash. But when giving someone cash you can't accidentally miss the person's hand causing the cash to disappear. At worst it falls on the floor and you pick it up.

Being technically able to send funds to an account that doesn't exist yet is necessary for offline wallets, but that is not the most common usage.

A few possibilities:

When you paste an address into the wallet, it immediately runs a check to see if the account exists. If the account doesn't exist then it says "account does not exist" and requires you to select one of:

- I'm activating an account.
- I'm sending to an offline wallet, yeah yeah I know what I'm doing quit bothering me.

If you select activating a new account, it sends a tiny amount of XRB and then checks again to see if a corresponding "open" transaction is created. Ideally it should be possible to open an account without sending anything (this would be neater, easier on the eyes). Once it sees the first block, it sends the rest.

**Comment by citizen2077** [comment]
I agree this sort of functionality would be a bonus to both noobs and seasoned crypto users

**Comment by androm3da** [comment]
IIUC, Raiblocks might be uniquely positioned to make this distinction.  "I'm creating the first transaction for the destination" might be feasible to determine for the sender.

Also note that something like bitcoins payment protocol (BIP70) might help here too.

**Comment by straightwhiteproud** [comment]
you're trying to create a solution for a nonexistent problem

**Comment by androm3da** [comment]
@straightwhiteproud I think you're mistaken if you think that has never happened before.  I don't think this needs to be the top-ranked issue but it seems like it could yield a good benefit for modest effort.

**Comment by fallerOfFalls** [comment]
Here's a [reddit thread](https://www.reddit.com/r/RaiBlocks/comments/7lwfd3/does_anyone_else_live_in_terror_of_inputting_the/) about it, for all the users who don't use GitHub (who are likely to be less technical). 

/u/Zarxrax has expressed the anxiety pretty well:
> Sending to the wrong address is a fear that I have every time. Even through its as simple as copy and paste, you always have to double check this long series of characters that is difficult to read. You think things like "what if I didn't get one of the characters in my selection", " what if it put an extra space at the end of my paste" (would that even cause an issue? I dont know) "what if I got a virus that will change the addresses that I paste". Sending transactions on every crypto are more stress inducing than they need to be.

**Comment by clemahieu** [comment]
I think there’s some merit to warning about sending to an address that has no activity and asking if someone is sure. 

The xrb_... address format has some pretty strong checks in place already to make sure you can’t mess it up as much as feasible. It’s prefixed with xrb so you know the currency it’s for. It doesnt allow extra or missing characters in case of a copy and paste issue, and it has a 40bit checksum to verify no letters were typed incorrectly.

**Comment by fallerOfFalls** [comment]
That's really interesting! However, users don't know that.

**Comment by fallerOfFalls** [comment]
@clemahieu I've done some tests now to confirm what you said. I had this idea that any combination of characters would be a valid address, but this is not the case.

So, it seems that wrong address *already* fails gracefully. But less technical users don't know this, and this will put them off. Is there a way that this can be made clear?

**Comment by clemahieu** [comment]
Sure, maybe we can add a tooltip giving an explanation.

**Comment by fallerOfFalls** [comment]
That would be totally amazing! How about something like [this](https://github.com/jaimehgb/RaiWebWallet/issues/13)?

**Comment by rkeene** [comment]
This appears to be a request for an option in the desktop wallet to warn the user if the destination account has no activity, which seems reasonable

**Comment by fallerOfFalls** [comment]
That summary seems almost entirely wrong. It's so wrong that I'm not sure where to start in correcting it. I mean, yes, that's a good idea, but not really the main issue.
